tpgtools/main.go

```diff
@@ -63,37 +65,39 @@ func main() {
 
 	if mode != nil && *mode == "serialization" {
 		generateSerializationLogic(resourcesForVersion)
-	} else {
```

Comment:
As an additional change, could it be even more clear if we use separate functions to handle serialization and provider generation?

Reply:
will do as a subsequent change